### PR TITLE
feat(sdk-router)!: add support for FastBridgeRouterV2

### PR DESCRIPTION
**Pull request by ChiTimesChi**
**Description**
Adds support for the updated `FastBridgeRouterV2` contract. This is a breaking change for the consumers previously using custom options like `deadline` or `excludeCCTP`. A single options object is now used instead:

https://github.com/synapsecns/sanguine/blob/d458ac13676fb681660ade31d2176b48ecb5b74e/packages/sdk-router/src/operations/bridge.ts#L60-L72

https://github.com/synapsecns/sanguine/blob/f4a762fc6c5ab48a69614def7de6cd3f9c49c0a3/packages/sdk-router/src/operations/bridge.ts#L135-L143

https://github.com/synapsecns/sanguine/blob/f4a762fc6c5ab48a69614def7de6cd3f9c49c0a3/packages/sdk-router/src/operations/bridge.ts#L91-L99

**Note:** integrations that require a smart contract to work (such as a smart wallet like Safe, or a bridge aggregator contract) **MUST** provide the `originUserAddress` option, otherwise the origin transaction will revert.

## Summary by CodeRabbit

- **New Features**
    - Introduced structured parameters for bridge-related functions to enhance clarity and usability.
    - Added a required `originUserAddress` parameter for improved handling of smart contract interactions in bridge operations.

- **Documentation**
    - Enhanced documentation to emphasize the necessity of `originUserAddress` when initiating bridge operations.

### CODE REVIEW

**Comment by ChiTimesChi** [comment]
Converting this to draft until the `FastBridgeRouterV2` is deployed

**Comment by ChiTimesChi** [comment]
The typescript tests are expected to pass once the FastBridgeRouterV2 is redeployed and this PR is updated with the correct address

File: packages/sdk-router/README.md [hunk 1]
```diff
@@ -77,17 +99,28 @@ const bridgeQuotes: BridgeQuote[] = await synapseSDK.allBridgeQuotes(
...
 )
 ```
 
-The returned list is sorted by the `maxAmountOut` field, so the first quote is the one yielding the highest amount of tokens on the destination chain.
-
-If either of the input/output tokens is a native gas token (e.g. ETH on Ethereum/Arbitrum, AVAX on Avalanche, etc.), the specialized address `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` should be used instead of the token address.
+- **Important:** It is crucial to provide the `originUserAddress` when a smart contract will initiate the bridge operation on behalf of the user. This requirement applies to smart wallets (e.g., Safe) and third-party integrations (such as bridge aggregator smart contracts).
+- For native gas tokens (e.g., ETH on Ethereum/Arbitrum, AVAX on Avalanche, etc.), use the specialized address `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` instead of the token address for either input or output tokens.
+- The returned list is sorted by the `maxAmountOut` field in descending order, with the first quote offering the highest amount of tokens on the destination chain.
+- All quotes in the list are provided without any slippage applied. To add slippage to the quotes, use the `applyBridgeSlippage` function.
```
Comment (Defi-Moses) [inline packages/sdk-router/README.md:118]: lets change the language to "without any slippage settings applied" saying "without any slippage" doesnt fully communicate the point imo

File: packages/sdk-router/README.md [hunk 2]
```diff
@@ -46,6 +46,28 @@ const synapseSDK = new SynapseSDK(chainIds, providers)
...
+### Router Deployments
+
+The Routers for each module are deployed to consistent addresses across all chains, with a few exceptions noted below.
+
+| Bridge Module | Chain   | Address                                      |
+| ------------- | ------- | -------------------------------------------- |
+| SynapseBridge | All[^1] | `0x7E7A0e201FD38d3ADAA9523Da6C109a07118C96a` |
+| SynapseBridge | Blast   | `0x0000000000365b1d5B142732CF4d33BcddED21Fc` |
+| SynapseCCTP   | All     | `0xd5a597d6e7ddf373a92C8f477DAAA673b0902F48` |
+| SynapseRFQ    | All     | `0x00cD000000003f7F682BE4813200893d4e690000` |
```
Comment (Defi-Moses) [inline packages/sdk-router/README.md:58]: are the fastbridge routers all at the same address even though the bridge contract is different on blast and linea?

Comment (ChiTimesChi) [reply packages/sdk-router/README.md:58]: Yes, the FastBridgeRouter has the same address regardless of FastBridge address

File: packages/sdk-router/README.md (hunk 1 above)
Comment (ChiTimesChi) [reply packages/sdk-router/README.md]: Fixed in 1813785dedee0b74265614df8cb9ae17cbb8ed7f

**Pull request merged**